### DISCUSSION TITLE
Neighboring Pixels

### DISCUSSION THREAD

**Discussion by natedailey7** [azavea/raster-vision#1955] [Q&A] [answered]
I have more of a conceptual question about Raster Vision. Does Raster Vision take into account neighboring pixel values when making a prediction? In contrast to classifying each pixel individually, I'm wondering if Raster Vision uses neighboring pixels as context. For instance, if identifying a tree, a dark shadow around the actual object would be considered, rather than just the green pixels of the tree canopy. I'm thinking... maybe the Object Detection classifier does this but Semantic Segmentation does not?

**Comment by AdeelH** [comment, marked as answer]
This is a question about model architectures rather than Raster Vision itself. [Convolutional model architectures](https://cs231n.github.io/convolutional-networks/) take neighboring pixels into account. Both Raster Vision's default model and the custom model used in the tutorials are convolutional networks.

**Reply by natedailey7** [reply]
That makes sense, thank you! Excited to dig deeper on this.
